assets/blocks/quiz/quiz-block/quiz-settings.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * WordPress dependencies
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+	PanelBody,
+	PanelRow,
+	RangeControl,
+	ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import NumberControl from '../../editor-components/number-control';
+
+/**
+ * Quiz settings.
+ *
+ * @param {Object}   props                    Block props.
+ * @param {Object}   props.attributes         Block attributes
+ * @param {Object}   props.attributes.options Current setting options.
+ * @param {Function} props.setAttributes      Set attributes function.
+ */
+const QuizSettings = ( { attributes: { options = {} }, setAttributes } ) => {
+	const {
+		passRequired = false,
+		quizPassmark = 100,
+		autoGrade = true,
+		allowRetakes = true,
+		randomQuestionOrder = false,
+		showQuestions = null,
+	} = options;
+
+	const createChangeHandler = ( optionKey ) => ( value ) =>
+		setAttributes( { options: { ...options, [ optionKey ]: value } } );
+
+	return (
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Quiz Settings', 'sensei-lms' ) }
+				initialOpen={ true }
+			>
+				<PanelRow>
+					<ToggleControl
+						checked={ passRequired }
+						onChange={ createChangeHandler( 'passRequired' ) }
+						label={ __( 'Pass Required', 'sensei-lms' ) }
```

Comment:
I think we could add a help text for this, something like `Require passing the quiz to complete lesson.`?

Similarly, **Passing Grade** needs some more clarification that it's actually a percentage.

cc @donnapep

Reply:
Help text: https://github.com/Automattic/sensei/pull/3987/commits/aad5685c6fb1417f64c9474a7f2d6ebb1dffda42

---

About the Passing Grade, I just added a (%) to the label. If we think it's not enough, we can go with a more complex solution adding a `%` signal over the input, similar to what you designed. I don't think it'd be ideal because depending on how the `RangeControl` component is updated in Gutenberg, it could break.

The `RangeControl` also supports an icon to be added before or after the bar, but I think in the label it makes more sense.

https://github.com/Automattic/sensei/pull/3987/commits/166bd57837738144fe2664950eb465b2695ae98e